packages/web-fragments/src/elements/reframed/script-execution.ts

```diff
@@ -61,15 +61,15 @@ export function executeScriptsInPiercedFragment(shadowRoot: ShadowRoot, iframe:
 		const iframeDocument = iframe.contentDocument;
 
 		assert(iframeDocument !== null, 'iframe.contentDocument is not defined');
-
-		getInternalReference(iframeDocument, 'body').appendChild(iframeDocument.importNode(script, true));
+		executeInertScript(script, iframeDocument);
 	});
 }
 
 /**
  * Weak map of scripts running in the iframe to their inert clones in the reframed DOM.
  */
 export const execToInertScriptMap = new WeakMap<HTMLScriptElement, HTMLScriptElement>();
+export const alreadyExecutedScripts = new Set<HTMLScriptElement>();
```

Comment:
Can you use WeakSet instead please? We don't want to retain memory unnecessarily.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WeakSet

Reply:
Yep, done